src/main/java/com/teragrep/pth_06/planner/offset/DatasourceOffset.java

```diff
@@ -63,10 +63,26 @@ public class DatasourceOffset extends Offset implements Serializable {
 
     SerializedDatasourceOffset serializedDatasourceOffset;
 
+    public DatasourceOffset(HdfsOffset hdfsOffset, LongOffset archiveOffset, KafkaOffset kafkaOffset) {
```

Comment:
I think this could be refactored, also relating to the if-else tree mentioned before, but perhaps it is for another issue

Reply:
Solved in commit 03552980d1289068474c778d099b2d0dc979c93f